electron/app/js/ipcRendererPreload.js

```diff
@@ -203,7 +203,8 @@ contextBridge.exposeInMainWorld(
           'get-wrc-home-directory',
           'get-wrc-app-image',
           'wrc-get-home-default-value',
-          'wrc-set-home-and-start'
+          'wrc-set-home-and-start',
+          'get-wrc-port'
```

Comment:
As we discussed a few weeks back, I don't understand the purpose of this IPC call.  The electron side is getting the port number and pushing it to the webui side once it is available.  Making the webui side pull the port number doesn't eliminate the asynchronous nature of this and your code still has to be prepared to get `undefined` back since your code may call this method before the backend has started and the port is known.

My sense is that you think you are eliminating the asynchrony because you haven't seen the race condition but rest assured that you are not eliminating the race condition by doing this.

Reply:
I was not thinking that the get-wrc-port IPC eliminated the race condition. In fact, lines 51-53 in the model-design-view.js model look for whether the port number sent to windowStateUtils.js was undefined, or not.

The get-wrc-port IPC is there because there scenarios involving the WRC, which could leave the WKT-UI/WRC piece in an unstable state. For instance, if the spawned WRC app dies (or literally gets killed by the end user) and the user is interacting with the "Design View" tab, then the get-wrc-port IPC is the way to gracefully recover. To see this, click on the "Design View" tab, kill the process associated with the started WRC app, then change a field in the WRC form or click on the navtree. The get-wrc-port IPC is the reason you end up back on the "WRC not installed" version of the "Design View" tab, and not dangling in a state where the embedded WRC is still trying to make REST calls to the CBE. You have it so the electron side prints the exit code of the killed WRC, but that doesn't 1) help the embedded WRC know that the backend is now invalid, and 2) set things up so that the "WRC not installed" version of the "Design View" tab gets displayed.

Reply:
Here's the stdout showing log entries from both the electron and webui side, in the scenario described above:

info: [renderer:2] showWdtModelDesigner: backendPort=52301
info: WebLogic Remote Console backend process exited with code 0
info: [renderer:2] showWdtModelDesigner: backendPort=undefined
info: [renderer:2] showWdtModelDesigner: backendPort=52340

The get-wrc-port IPC is what's allowing the model-design-view.js to figure out what's going on, and communicate the valid backendPort to the wdt-model-designer JET composite, so it can in turn set the valid backendURL to be used when doing CBE REST calls.

Reply:
So rather than this elaborate scheme that requires a new IPC every time the user switches to the Design View tab, why wouldn’t we just make the call to push the new value of the port when the process exits?